src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/SubaccountSupervisor.kt

```diff
@@ -569,216 +540,226 @@ internal class SubaccountSupervisor(
-                        transactionCallback(placeOrderResponse, uiDelayTimeMs, submitTimeMs)
-                    }
-                } else {
-                    helper.send(error, callback, payload)
+        val orderTransactionCallback = { response: String? ->
+            val error = parseTransactionResponse(response)
+            trackOrderSubmitted(error, analyticsPayload)
+            if (error == null) {
+                lastOrderClientId = clientId
+            } else {
+                val placeOrderRecord = this.placeOrderRecords.firstOrNull {
+                    it.clientId == clientId
                 }
+                this.placeOrderRecords.remove(placeOrderRecord)
             }
+            helper.send(error, callback, payload)
+        }
 
-        if (isShortTermOrder) {
-            val submitTimeMs = Clock.System.now().toEpochMilliseconds().toDouble()
-            val uiDelayTimeMs = submitTimeMs - uiClickTimeMs
-            if (isIsolatedMarginOrder) {
-                val transferPayload = getTransferPayloadForIsolatedMarginTrade(payload)
-                val transferPayloadString = Json.encodeToString(transferPayload)
 
-                helper.transaction(TransactionType.SubaccountTransfer, transferPayloadString) {
-                        response ->
-                    isolatedMarginTransactionCallback(response, uiDelayTimeMs, submitTimeMs)
-                }
-            } else {
-                helper.transaction(TransactionType.PlaceOrder, string) {
-                        response ->
-                    transactionCallback(response, uiDelayTimeMs, submitTimeMs)
-                }
-            }
-        } else {
-            if (isIsolatedMarginOrder) {
-                val transferPayload = getTransferPayloadForIsolatedMarginTrade(payload)
-                val transferPayloadString = Json.encodeToString(transferPayload)
-
-                transactionQueue.enqueue(
-                    TransactionParams(
-                        TransactionType.SubaccountTransfer,
-                        transferPayloadString,
-                        isolatedMarginTransactionCallback,
-                    ),
+        // If the transfer is successful, place the order
+        val isolatedMarginTransactionCallback = { response: String? ->
+            val error = parseTransactionResponse(response)
+            if (error == null) {
+                submitTransaction(
+                    TransactionType.PlaceOrder,
+                    string,
+                    onSubmitOrderTransaction,
+                    orderTransactionCallback,
+                    isShortTermOrder,
                 )
             } else {
-                transactionQueue.enqueue(
-                    TransactionParams(
-                        TransactionType.PlaceOrder,
-                        string,
-                        transactionCallback,
-                    ),
-                )
+                helper.send(error, callback, payload)
```

Comment:
should this be the transferPayload or just orderPayload? @jaredvu

Reply:
umm seems like jared hasn't incorporated it to v4-web so yea i'll change to transfer for now.. it used to be the order payload tho